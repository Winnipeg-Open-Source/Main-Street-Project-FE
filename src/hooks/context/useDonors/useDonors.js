import { useContext } from 'react';
import DonorsContext from 'context/Donors';

function useDonors () {
    return useContext(DonorsContext);
}

export default useDonors;
