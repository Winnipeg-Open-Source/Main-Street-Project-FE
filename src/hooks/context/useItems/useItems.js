import { useContext } from 'react';
import ItemsContext from 'context/Items';

function useItems () {
    return useContext(ItemsContext);
}

export default useItems;
