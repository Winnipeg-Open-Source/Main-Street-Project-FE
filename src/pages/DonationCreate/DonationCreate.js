import React, { useEffect } from 'react';
import Page from 'components/Page';
import DonationForm from 'containers/DonationForm';
import useDonors from 'hooks/context/useDonors';
import useItems from 'hooks/context/useItems';

function DonationCreate () {
    const { isLoading: isLoadingDonors, onLoadDonors } = useDonors();
    const { isLoading: isLoadingItems, onLoadItems } = useItems();

    useEffect(() => {
        isLoadingDonors && onLoadDonors();
        isLoadingItems && onLoadItems();
    }, []);

    return (
        <Page title='New Donation'>
            <DonationForm />
        </Page>
    );
}

DonationCreate.displayName = 'DonationCreate';

export default DonationCreate;
