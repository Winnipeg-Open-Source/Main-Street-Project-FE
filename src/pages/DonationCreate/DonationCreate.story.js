import React from 'react';
import StorybookRouter from 'storybook/helpers/StorybookRouter';
import MockProviders from 'storybook/helpers/MockProviders';
import mockAxios from 'storybook/helpers/mockAxios';
import DonationCreate from './DonationCreate';
import { multipleDonors } from 'storybook/mocks/donors';
import { mockItems } from 'storybook/mocks/items';
import { DONORS_API_PATH, ITEMS_API_PATH } from 'constants/api';

export const EmptyForm = () => {
    mockAxios.onGet(DONORS_API_PATH).reply(200, multipleDonors.donors);
    mockAxios.onGet(ITEMS_API_PATH).reply(200, mockItems);

    return (
        <StorybookRouter>
            <MockProviders>
                <DonationCreate />
            </MockProviders>
        </StorybookRouter>
    );
};

export default {
    title: 'Pages/DonationCreate',
    component: DonationCreate,
};
