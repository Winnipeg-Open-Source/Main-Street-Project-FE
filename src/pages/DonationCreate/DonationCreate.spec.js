import React from 'react';
import renderWithThemeAndRouter from 'tests/helpers/renderWithThemeAndRouter';
import DonationCreate from './DonationCreate';
import useDonors from 'hooks/context/useDonors';
import useItems from 'hooks/context/useItems';

jest.mock('hooks/context/useDonors');
jest.mock('hooks/context/useItems');

describe ('DonationCreate', () => {
    it ('renders without crashing', () => {
        const mockLoadDonors = jest.fn();
        const mockLoadItems = jest.fn();

        useDonors.mockImplementation(() => ({ isLoading: true, onLoadDonors: mockLoadDonors }));
        useItems.mockImplementation(() => ({ isLoading: true, onLoadItems: mockLoadItems }));

        const { asFragment } = renderWithThemeAndRouter(<DonationCreate />);

        expect(mockLoadDonors).toHaveBeenCalledTimes(1);
        expect(mockLoadItems).toHaveBeenCalledTimes(1);
        expect(asFragment()).toMatchSnapshot();
    });

    it ('does not load already loaded resources', () => {
        const mockLoadDonors = jest.fn();
        const mockLoadItems = jest.fn();

        useDonors.mockImplementation(() => ({ isLoading: false, onLoadDonors: mockLoadDonors }));
        useItems.mockImplementation(() => ({ isLoading: false, onLoadItems: mockLoadItems }));

        renderWithThemeAndRouter(<DonationCreate />);

        expect(mockLoadDonors).toHaveBeenCalledTimes(0);
        expect(mockLoadItems).toHaveBeenCalledTimes(0);
    });
});
