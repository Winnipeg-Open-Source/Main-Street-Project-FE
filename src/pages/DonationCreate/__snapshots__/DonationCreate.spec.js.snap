// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonationCreate renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="sc-pbJYR hNzbLI sc-pkfPC fQMEfn"
  >
    <div
      class="sc-AxjAm sc-AxiKw bENblX"
      width="1"
    >
      <div
        class="sc-AxjAm sc-AxiKw iPzLAe"
        width="1"
      >
        <div
          class="sc-AxhCb sc-psDhf lcjizo"
          font-size="3"
        >
          New Donation
        </div>
      </div>
      <div
        class="sc-AxjAm sc-AxiKw sc-qYUWV dgGtKY"
      >
        <div
          class="sc-AxjAm sc-AxiKw sc-pBlxj iKhLIT"
        >
          <label
            class="sc-oVpqz cpWgdv className"
            color="border.light"
            font-size="10px"
            font-weight="bold"
          >
            <div
              class="sc-AxjAm sc-AxiKw iSVrPI"
              width="1"
            >
              <select
                class="sc-pZCuu kQVKTB sc-oTaid sc-pRgDJ dojhwH"
                data-testid="employee-select"
                disabled=""
                font-size="2,,1"
                id="Donor"
              >
                <option
                  disabled=""
                  hidden=""
                  value=""
                >
                  Loading...
                </option>
              </select>
              <svg
                aria-hidden="true"
                class="sc-Axmtr gXMRoh sc-fzqPZZ sc-pRFHb JnCiE"
                color="text.light"
                fill="currentcolor"
                focusable="false"
                height="24"
                role="img"
                tabindex="-1"
                viewBox="0 0 24 24"
                width="24"
              >
                <path
                  d="M7.4 8l4.6 4.6L16.6 8 18 9.4l-6 6-6-6L7.4 8z"
                />
              </svg>
            </div>
          </label>
          <input
            class="sc-pTWqp sc-pJipy ihfsBt"
            font-size="2,,1"
            placeholder="Title"
            value=""
          />
          <label
            class="sc-oVpqz CTGsT"
            color="border.light"
            font-size="1"
            font-weight="bold"
          >
            Description
            <textarea
              class="sc-qYhTA sc-pRtcU HjXGg"
              id="donation-description"
            />
          </label>
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="downshift-0-label"
            role="combobox"
            style="position: relative; width: 100%;"
          >
            <input
              aria-autocomplete="list"
              aria-labelledby="downshift-0-label"
              autocomplete="off"
              class="sc-pTWqp fyIUKN sc-pbvYO hcYciJ"
              disabled=""
              font-size="2,,1"
              id="downshift-0-input"
              placeholder="Loading..."
              value=""
              width="1"
            />
            <div
              aria-labelledby="downshift-0-label"
              class="sc-AxjAm sc-qYsuA sc-pZdvY fOXuV"
              height="256px"
              id="downshift-0-menu"
              role="listbox"
            />
          </div>
          <div
            class="sc-AxjAm sc-AxiKw sc-pjumZ jcAEGC"
            width="1"
          />
          <label
            class="sc-oVpqz cjcRzm"
            color="border.light"
            font-size="1"
            font-weight="bold"
          >
            Additional Notes
            <textarea
              class="sc-qYhTA sc-pRtcU gvsFZp"
              id="donation-notes"
            />
          </label>
        </div>
        <div
          class="sc-AxjAm sc-AxiKw sc-qQxFs cuUChO"
        >
          <button
            class="sc-oUOMp kxZVgg"
            color="tertiary"
            width="1"
          >
            Cancel
          </button>
          <button
            class="sc-oUOMp coOwBP"
            color="primary"
            width="1"
          >
            Save
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
